core/src/test/java/org/elasticsearch/indices/cluster/ClusterStateChanges.java

```diff
@@ -119,7 +122,7 @@ public ClusterStateChanges() {
         IndexNameExpressionResolver indexNameExpressionResolver = new IndexNameExpressionResolver(settings);
         DestructiveOperations destructiveOperations = new DestructiveOperations(settings, clusterSettings);
         Environment environment = new Environment(settings);
-        ThreadPool threadPool = null; // it's not used
+        threadPool = new TestThreadPool("ClusterStateChanges");
```

Comment:
I would prefer that we don't use a threadpool in this class, it's supposed to be side-effect free. I prefer to set request.waitForActiveShards() to NONE and make a shortcut in MetaDataCreateIndexService for NONE.